repose-aggregator/components/services/container-configuration-service/impl/src/test/scala/org/openrepose/nodeservice/containerconfiguration/ContainerConfigurationServiceImplTest.scala

```diff
@@ -0,0 +1,272 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.nodeservice.containerconfiguration
+
+import java.net.URL
+
+import org.junit.runner.RunWith
+import org.mockito.Mockito.{never, verify, when}
+import org.mockito.{Matchers => MockitoMatchers}
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.core.container.config._
+import org.openrepose.core.services.config.ConfigurationService
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FunSpec, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class ContainerConfigurationServiceImplTest extends FunSpec with Matchers with MockitoSugar with BeforeAndAfterEach {
+
+  final val DefaultClusterId = "defaultClusterId"
+
+  var mockConfigService: ConfigurationService = _
+  var containerConfigurationService: ContainerConfigurationServiceImpl = _
+
+  override def beforeEach() = {
+    mockConfigService = mock[ConfigurationService]
+    containerConfigurationService = new ContainerConfigurationServiceImpl(DefaultClusterId, mockConfigService)
+  }
+
+  describe("init") {
+    it("should register configuration listener") {
+      containerConfigurationService.init()
+
+      verify(mockConfigService).subscribeTo(
+        MockitoMatchers.eq(ContainerConfigurationServiceImpl.ContainerConfigurationFilename),
+        MockitoMatchers.any[URL](),
+        MockitoMatchers.any[UpdateListener[ContainerConfiguration]](),
+        MockitoMatchers.any[Class[ContainerConfiguration]]()
+      )
+    }
+  }
+
+  describe("destroy") {
+    it("should unregister configuration listener") {
+      containerConfigurationService.destroy()
+
+      verify(mockConfigService).unsubscribeFrom(
+        MockitoMatchers.eq(ContainerConfigurationServiceImpl.ContainerConfigurationFilename),
+        MockitoMatchers.any[UpdateListener[ContainerConfiguration]]()
+      )
+    }
+  }
+
+  describe("isInitialized") {
+    it("should return false if a configuration has not yet been read") {
+      containerConfigurationService.isInitialized shouldBe false
+    }
+
+    it("should return true if a configuration has been read") {
+      containerConfigurationService.configurationUpdated(minimalContainerConfiguration())
+
+      containerConfigurationService.isInitialized shouldBe true
+    }
+  }
+
+  describe("getVia") {
+    it("should throw an Exception if the service is not yet initialized") {
+      intercept[IllegalStateException] {
+        containerConfigurationService.getVia
+      }
+    }
+
+    it("should return an empty Optional if not configured") {
+      containerConfigurationService.configurationUpdated(minimalContainerConfiguration())
+
+      containerConfigurationService.getVia.isPresent shouldBe false
+    }
+
+    it("should return the configured via string") {
```

Comment:
This should be two tests, one to make sure you can get the base value, and one to make sure you can get a patched value.